core/src/main/java/overflowdb/util/DummyEdgeIterator.java

```diff
@@ -17,6 +17,9 @@
   private final String label;
   private final NodeRef thisRef;
 
+  /** used for peeking */
+  private Edge nextCached;
+
   public DummyEdgeIterator(Object[] array, int begin, int exclusiveEnd, int strideSize,
                            Direction direction, String label, NodeRef thisRef) {
```

Comment:
Let's document that this is a benign race condition (`array` can be modified under us, but this is OK and we spent time thinking about the fact).

Reply:
done